### PR TITLE
REG-34: Stop skipping credit donation test

### PR DESCRIPTION
**Pull request by bdsl**
I expect this will fail, but pushing to Circle CI to see how it fails initially.

### CODE REVIEW

**Review by dorota-joanna — APPROVED** [review]
LGTM

File: features/donation-with-credits.feature [hunk 1]
```diff
@@ -18,7 +17,7 @@ Feature: Existing donor: credit donation completes successfully
         Then I should see my populated first name is "RegressionTest"
         And I should see my populated surname is "User"
         And I should see my populated email is "tech+regression+credits@thebiggive.org.uk"
-        And I should see "Your credit balance will be applied against this donation. No further funds will be taken." instead of asking for my bank details.
+        And I should see "Your donation funds balance will be applied against this donation. No further funds will be taken." instead of asking for my bank details.
```
Comment (bdsl) [inline features/donation-with-credits.feature:20]: Looks like this copy change is the only reason this test was failing until now and so had to be skipped.

File: .circleci/config.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
           paths:
             - ~/.cache
       - run: yarn lint
-      - run: yarn tsc || echo "Typescript errors found, not failing build"
+      - run: yarn tsc
```
Comment (bdsl) [inline .circleci/config.yml:28]: from now on if we write any code that doesn't work for the typescript compiler we'll get a build error in circle and the actual tests won't run. Much quicker and more reliable feedback than waiting for the tests to run, and forces us to keep the some of the inline documentation more accurate.

File: pages/DonateStartPage.js [hunk 3]
```diff
@@ -68,7 +68,7 @@ export default class DonateStartPage {
      * while keeping the step definitions simple.
      *
      * @param {string} selector Element selector.
-     * @returns {any} click() result on success.
+     * @returns {Promise<any>} click() result on success.
```
Comment (bdsl) [inline pages/DonateStartPage.js:71]: most changes are fixes for errors the typescript compiler found. E.g. async functions always return promises.

File: support/check.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { WAIT_SECONDS } from './constants';
  * Check if element exists
  *
  * @param {string} selector to be checked
- * @param {int} seconds to wait
+ * @param {number} seconds to wait
```
Comment (bdsl) [inline support/check.js:7]: no such thing as an `int` type in Javascript.

**Review by dorota-joanna — COMMENTED** [review]
LGTM

**Pull request merged**